File: Go/Creational/Singleton/Example3/singleton/singleton.go
package singleton

import (
	"sync"
)

type Singleton struct {
	Number      int
	Description string
}

var (
	instance *Singleton
	once     sync.Once
)

func GetInstance() *Singleton {
	once.Do(func() {
		instance = &Singleton{}
	})
	return instance
}

File: Swift/Creational/Singleton/Singleton/Singleton.swift
import Foundation

class Singleton: CustomStringConvertible {
    // class constant with lazy initialization
    // also thread safe
    static let sharedInstance = Singleton()
    
    var number: Int = 0
    var someDescription: String = ""
    
    private init() {}
    
    var description: String {
        return "Singleton{number:\(self.number), someDescription:\(self.someDescription)}"
    }
}
